crates/bevy_ecs/src/event.rs

```diff
@@ -313,7 +312,7 @@ impl<'w, 's, E: Event> EventWriter<'w, 's, E> {
         self.events.send(event);
     }
 
-    pub fn send_batch(&mut self, events: impl Iterator<Item = E>) {
+    pub fn send_batch(&mut self, events: impl IntoIterator<Item = E>) {
```

Comment:
This needs a note in the migration guide.

Reply:
Because of following blanket impl in `std` of `IntoIterator` on every `Iterator` this is not a breaking change AFAIK.
```rust
impl<I: Iterator> IntoIterator for I {
    type Item = I::Item;
    type IntoIter = I;

    #[inline]
    fn into_iter(self) -> I {
        self
    }
}
```